test/dryrun/context_vars/object_pod_nsselector/error.txt

```diff
@@ -0,0 +1,7 @@
+Error:
+unable to complete the dryrun reconcile:
+failed to resolve the template {"apiVersion":"v1","kind":"Pod","metadata":{"labels":{"image":"{{ index .Object.spec.containers 0 \"image\" }}"},"name":"nginx-pod"}}:
+template:
+tmpl:5:28:
+executing "tmpl" at <.Object.spec.containers>:
+can't evaluate field Object in type struct { ObjectNamespace string; ObjectName string }
```

Comment:
I suppose it makes sense to have an error if you use `.Object` in a template when the resource doesn't exist yet... but it also feels like a bit of a major limitation. Is there a way for the user to provide a default value when the resource doesn't exist yet?

Reply:
I've updated the logic to populate `.Object` in all cases except for unnamed `objectDefinition`s. Please take another look when you have a minute!